### PR TITLE
[homekit] implement List-Pairings method

### PR DESCRIPTION
**Pull request by ccutrer**
refs #13949

### CODE REVIEW

**Comment by ccutrer** [comment]
I have not test this at all yet, and it will need to use the released HAP-java once I have, I just wanted to get the code up. I don't have much time right now.

**Review by yfre — APPROVED** [review]
LGTM

File: bundles/org.openhab.io.homekit/pom.xml [hunk 1]
```diff
@@ -22,7 +22,7 @@
     <dependency>
       <groupId>io.github.hap-java</groupId>
       <artifactId>hap</artifactId>
-      <version>2.0.4</version>
+      <version>2.0.5</version>
```
Comment (jlaur) [inline bundles/org.openhab.io.homekit/pom.xml:25]: Is it of concern that 2.0.4 is listed as latest [here](https://github.com/hap-java/HAP-Java/releases)?

**Comment by jlaur** [comment]
@ccutrer - is this PR confirmed to fix #13949? If so, I would assume that the merge commit should be cherry-picked into the 3.4.x hotfix branch, but please confirm.

File: bundles/org.openhab.io.homekit/pom.xml (hunk 1 above)
Comment (ccutrer) [reply bundles/org.openhab.io.homekit/pom.xml:25]: Not really. There's an issue with our GitHub action doing the release that it doesn't always create that for some reason. The artifact is on maven central through: https://mvnrepository.com/artifact/io.github.hap-java/hap

**Comment by ccutrer** [comment]
> @ccutrer - is this PR confirmed to fix #13949? If so, I would assume that the merge commit should be cherry-picked into the 3.4.x hotfix branch, but please confirm.

Yeah, I'm pretty confident in it now. I'm ready to have this merged assuming someone will approve the code review. ~48 hours of lots of good experiences on that issue and the community thread. If we could also get https://github.com/openhab/openhab-addons/pull/14017 cherry-picked over too - it's a minor fix, but can prevent some head scratching if you are having issues with HomeKit pairing.

**Review by jlaur — APPROVED** [review]
Thanks for all the effort resolving this issue.

**Pull request merged**